src/Microsoft.Extensions.Configuration.AzureAppConfiguration/IConfigurationClientManager.cs

```diff
@@ -5,12 +5,13 @@
 using Azure.Data.AppConfiguration;
 using System;
 using System.Collections.Generic;
+using System.Threading;
 
 namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
 {
     internal interface IConfigurationClientManager
     {
-        IEnumerable<ConfigurationClient> GetAvailableClients(DateTimeOffset time);
+        IEnumerable<ConfigurationClient> GetAvailableClients();
 
         IEnumerable<ConfigurationClient> GetAllClients();
```

Comment:
These two are odd. Separate method for a subset of data is not great for an interface. 
In this case we can always return all and have a property in ConfigurationClient that the caller can filter on instead.

Reply:
The ConfigurationClient type is not defined in this project, it's from .net SDK, we can't add a property to it, we can only extend it or wrap it.

Then it seems would be a little bit overkill since the callers only care about the clients, they don't have to worry about how the clients are provided.

Reply:
From my perspective , changing this seems doesn't make it much better.
https://github.com/Azure/AppConfiguration-DotnetProvider/pull/502/files

Reply:
What's the reason we need two different subsets of ConfigurationClients (Available vs All)?

Reply:
`GetAllClients` is a new method since 7.0.0 for time-based retry on start-up.  When the app starts, there is no longer a fixed number of retries for each client, but as long as the time allows (default is 100s), we will continue to try all clients until a certain client can work or the time runs out. Therefore, it is necessary to obtain all Clients. Of course, we will not keep trying all Clients without limit. Whenever a round of attempts is made and all Clients are invalid, it will back off for a certain period of time and then start trying the next round.

`GetAvailableClients` is for getting the available clients for refreshing scenario, each client has unique backoff time which is determined by its failure attempts. Hence, we would always get the clients whose back off time is due.

Reply:
I think we should have uniform strategy to use client instances in all cases (startup and refresh). Beside on startup, backoff isn't specified (no one has failed yet).  I would assume that at startup GetAvailableClients always return all clients. Am I missing something?

cc @jimmyca15

Reply:
Good idea, I would merge this one first, go out in 7.1.0-preview first.